### PR TITLE
Seperate segmented algorithms for reduce

### PR DESCRIPTION
**Pull request by Jedi18**
Seperate segmented algorithms for reduce
Replace tag_invoke with tag_fallback_invoke
Add sentinel tests for reduce.

## Any background context you want to provide?
Issue #5156
Issue #5204

### CODE REVIEW

**Comment by hkaiser** [comment]
retest

**Comment by msimberg** [comment]
retest

File: libs/full/segmented_algorithms/include/hpx/parallel/segmented_algorithms/reduce.hpp [hunk 1]
```diff
@@ -186,36 +186,74 @@ namespace hpx { namespace parallel { inline namespace v1 {
...
+            "Requires at least input iterator.");
+
+        static_assert(hpx::traits::is_input_iterator<InIterE>::value,
+            "Requires at least input iterator.");
+
+        if (first == last)
         {
-            typedef hpx::is_sequenced_execution_policy<ExPolicy> is_seq;
-            typedef typename std::decay<T>::type init_type;
+            return std::forward<T>(init);
```
Comment (hkaiser) [inline libs/full/segmented_algorithms/include/hpx/parallel/segmented_algorithms/reduce.hpp:216]: No need to `std::forward` the argument. It was passed by value.

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Pull request merged**